indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -388,16 +760,191 @@ public void shutdown(final String taskid)
           @Override
           public InputStream openStream() throws IOException
           {
-            return LogUtils.streamFile(processHolder.logFile, offset);
+            return LogUtils.streamFile(logFile, offset);
           }
         }
     );
   }
 
+  private void populateMissingTasksFromDir()
+  {
+    if(started.get() || stopped.get())
+    {
+      // This might be safe to do, but this method assumes this is not the case
+      throw new ISE("Cannot populate tasks from dirs once ForkingTaskRunner has been started");
+    }
+    final File baseDir = taskConfig.getBaseTaskDir();
+    final File[] taskDirFileArray = baseDir.listFiles();
+    final Collection<File> taskDirFileList =
+        taskDirFileArray == null ?
+        ImmutableList.<File>of() :
+        Collections2.filter(
+            Arrays.asList(
+                taskDirFileArray
+            ), new Predicate<File>()
+            {
+              @Override
+              public boolean apply(File input)
+              {
+                return input.exists() && input.isDirectory();
+              }
+            }
+        );
+    if (taskDirFileList.isEmpty()) {
+      log.info("No task dirs found in [%s]", baseDir);
+      return;
+    }
+    for (File potentialTaskDir : taskDirFileList) {
+      final File[] taskAttemptDirFileArray = potentialTaskDir.listFiles();
+      final Collection<File> taskAttemptDirFileList =
+          taskAttemptDirFileArray == null ?
+          ImmutableList.<File>of() :
+          Collections2.filter(
+              Arrays.asList(taskAttemptDirFileArray),
+              new Predicate<File>()
+              {
+                @Override
+                public boolean apply(File input)
+                {
+                  return input.isDirectory() && input.listFiles(
+                      new FilenameFilter()
+                      {
+                        @Override
+                        public boolean accept(File dir, String name)
+                        {
+                          return TASK_FILE_NAME.equals(name)
+                                 && new File(dir, TASK_FILE_NAME).length()
+                                    > 0;
+                        }
+                      }
+                  ) != null;
+                }
+              }
+          );
+      if (taskAttemptDirFileList.isEmpty()) {
+        log.info("Directory [%s] has no viable task attempts, attempting to cleanup if empty");
+        if (!potentialTaskDir.delete()) {
+          log.warn("Could not clean up [%s]", potentialTaskDir.toString());
+        }
+        continue;
+      }
+      File latestAttemptDir = null;
+      for (File taskAttemptDir : taskAttemptDirFileList) {
+        if (latestAttemptDir == null) {
+          latestAttemptDir = taskAttemptDir;
+        } else {
+          if (latestAttemptDir.lastModified() < taskAttemptDir.lastModified()) {
```

Comment:
The attempt id is not a number, it is (and was) a uuid. I assume this is so that any particular task can go in and have a new attempt run while the old logs and data still persist.

I can change it to an ordinal directory naming scheme if attempt can be taken to mean "attempt in this particular directory structure" rather than "global attempt"